yaml/com/ibm/Panel.interface.yaml

```diff
@@ -0,0 +1,85 @@
+description: >
+    Implements control panel activities.
+methods:
+    - name: Display
+      description: >
+          Commands the op-panel to display the given lines.
+      parameters:
+          - name: line1
+            type: string
+            description: >
+                First line that needs to be displayed in op-panel LCD.
+          - name: line2
+            type: string
+            description: >
+                Second line that needs to be displayed in op-panel LCD.
+
+    - name: TriggerPanelLampTest
+      description: >
+          Commands the op-panel to trigger panel lamp test.
+      parameters:
+          - name: status
+            type: boolean
+            description: >
+                Panel lamp test request.
+
+    - name: toggleFunctionState
+      description: >
+          Toggles the state of given panel functions.
+      parameters:
+          - name: bitmap
+            type: array[byte]
+            description: >
+                Bit number in bitmap will have one to one mapping with function
+                number. Bit value 1 to enable and 0 to disable the function.
+
+    - name: ProcessButton
+      description: >
+          Method which simulates panel button request.
+      parameters:
+          - name: event_id
+            type: int32
+            description: >
+                Event ID = 0 simulates UP button. Event ID = 1 simulates DOWN
+                button. Event ID = 2 simulates ENTER button.
+
+    - name: ExecuteFunction
+      description: >
+          Trigger panel function directly.
+      parameters:
+          - name: function_number
+            type: byte
+            description: >
+                Execute the given function number. Supports to trigger only IBMi
+                functions {21, 22, 34, 65, 66, 67, 68, 69, 70}.
+      returns:
+          - name: output
+            type: struct[boolean, string, string]
+            description: >
+                Tuple with values return code, panel display line1, panel
+                display line2, if the execution is successful.
+      errors:
+          - xyz.openbmc_project.Common.Error.InternalFailure
+          - xyz.openbmc_project.Common.Error.NotAllowed
+
+    - name: getEnabledFunctions
+      description: >
+          Get list of enabled functions.
+      returns:
+          - name: enabled functions list
+            type: array[byte]
+            description: >
+                List of enabled functions from the set of IBMi functions list
```

Comment:
Same here, get rid of IBMi and function numbers.
Just "List of enabled functions" is sufficient.

Reply:
Generic work of this function is to return a list of enabled functions. In future if there is a requirement to add more functions to this other than IBMi then we can do that. We will not have to modify this document.

Reply:
As your function name nowhere suggests it is limited to IBMi function, similarly its description should not.

Reply:
We can't express the whole intent in the function name itself. thats why we have a document to describe the function. From the description people can know what this function does. whenever requirement changes, we can update it.
@jinuthomas ^ any thoughts on this?

Reply:
There was a similar discussion at the time of function implementation as well? The whole target is to keep the things as generic as possible.
https://github.com/ibm-openbmc/ibm-panel/pull/190#discussion_r1265394738